src/Git.jl

```diff
@@ -63,9 +63,9 @@ function iscommit(sha1s::Vector; dir="")
 end
 
 immutable State
-    head::String
-    index::String
-    work::String
+    head::AbstractString
```

Comment:
Compat is already a dependency of most packages, and it's very lightweight. So it's almost always better to use it rather than reinventing version checks.